rust/connlib/tunnel/src/device_channel/tun_windows.rs

```diff
@@ -199,3 +208,30 @@ fn start_recv_thread(
         tracing::debug!("recv_task exiting gracefully");
     })
 }
+
+/// Sets MTU on the interface
+/// TODO: Set IP and other things in here too, so the code is more organized
+fn set_iface_config(iface_idx: u32, mtu: u32) -> Result<()> {
+    tracing::debug!("set_iface_config...");
+    let mut row = MIB_IPINTERFACE_ROW {
+        InterfaceIndex: iface_idx,
+        ..Default::default()
+    };
+
+    unsafe { GetIpInterfaceEntry(&mut row) }?;
+    row.ManagedAddressConfigurationSupported = BOOLEAN(0);
+    row.OtherStatefulConfigurationSupported = BOOLEAN(0);
+    // We use this to get/set the MTU and metric, family should be irrelevant
+    row.Family = AF_INET;
+    row.NlMtu = mtu;
+
+    // Force lowest possible metric so Windows will always choose us
+    row.UseAutomaticMetric = BOOLEAN(0);
+    row.Metric = 0;
+
+    // Ignore error if we can't set everything
+    unsafe { SetIpInterfaceEntry(&mut row) }?;
```

Comment:
How expected is this to happen? In these early days of testing the Windows client we may want to welcome crashes moreso than not since we'll get a better sense of the edges and variety of our installbase. Another option would be to call onDisconnection eagerly with this error.

I left a similar comment for the Apple client as well. Though for Apple we get a nice automated crash dump which is another reason to "let it crash" while we're in beta

Reply:
Oh yeah how does log uploading work? If we have a non-panic error and write it to the logs before crashing, will it be picked up on the cloud and tagged for us to look at somehow?

Reply:
I'll see if it's easy to enroll in Microsoft's crash reporting but I've never used it before

Reply:
@ReactorScram Currently if there's a crash it won't be logged. Only the tracing logs from within `connlib` (not the app itself) get automatically uploaded to GCP. The idea was that we'd be able to pilfer through them to uncover patterns in connectivity issues among clients.

For UI logs, the idea was that they'd be written somewhere and "Export logs" would zip them up to send to us. See #2721  

For crash logs, we don't have a cross-platform mechanism in place. We get reports "for free" for Android and Apple by virtue of their distribution system. Maybe there's something similar for Microsoft, or if not, we log them like we do for App UI logs so that "Export logs" will grab them.

Reply:
I actually haven't figured out how to separate the GUI logs from the connlib logs. Since we're setting a global subscriber in `tracing`, something would have to change. In cases where the same thread calls from the GUI into connlib, we'd have to err on the side of considering it GUI logs since those are not uploaded automatically, right?

And in general I assume we should try to catch errors instead of panicking, but if we don't have any idea how to handle an error, we should log it and exit the app as if we had panicked? (since panics skip graceful shutdown and are sort of a last resort)

I did found a `tracing-panic` crate that would allow us to log panics:

```
{"time":"2024-01-02T17:34:16.8338364Z","target":"firezone_windows_client::client::debug_commands","logging.googleapis.com/sourceLocation":{"file":"windows-client\\src-tauri\\src\\client\\debug_commands.rs","line":"13"},"severity":"INFO","message":"started log (DebugCrash)"}
{"time":"2024-01-02T17:34:16.8343845Z","target":"tracing_panic","logging.googleapis.com/sourceLocation":{"file":"C:\\Users\\User\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\tracing-panic-0.1.1\\src\\lib.rs","line":"66"},"severity":"ERROR","message":"A panic occurred","panicLocation":"windows-client\\src-tauri\\src\\client\\debug_commands.rs:15:5","panicPayload":"purposely crashing to see if it shows up in logs"}
```

Windows does have a way to report errors through the OS. We'd need to get a special kind of certificate (probably costs money) and use that to sign the exe. After that it's free, and we get a little dashboard with crash / hang info. https://learn.microsoft.com/en-us/windows/win32/appxpkg/windows-desktop-application-program#health-report

Reply:
Hm not sure just yet, but opened https://github.com/firezone/firezone/issues/3111 to track this discussion further.